src/main/java/uk/gov/justice/digital/delius/service/ReferralService.java

```diff
@@ -1,46 +1,85 @@
 
-    @Autowired
-    public ReferralService(@Qualifier("deliusApiWebClient") final WebClient webClient) {
-        this.webClient = webClient;
+    NsiService nsiService;
+
+    OffenderService offenderService;
+
+    public ReferralService(final DeliusApiClient deliusApiClient, final NsiService nsiService, final OffenderService offenderService) {
+        this.deliusApiClient = deliusApiClient;
+        this.nsiService = nsiService;
+        this.offenderService = offenderService;
     }
 
     @Transactional
-    public ResponseEntity<String> createReferralSent(final String crn,
-                                                     final ReferralSentRequest referralSent) {
+    public ReferralSentResponse createNsiReferral(final String crn,
+                                                  final ReferralSentRequest referralSent) {
+        var existingNsi = getExistingMatchingNsi(crn, referralSent);
+
+        if(existingNsi.isPresent()) {
+            return ReferralSentResponse.builder().nsiId(existingNsi.get().getNsiId()).build();
+        }
 
-        var contact = NewContact.builder()
+        var newNsiRequest = NewNsi.builder()
+            .type(referralSent.getNsiType())
+            .subType(referralSent.getNsiSubType())
             .offenderCrn(crn)
-            .type(referralSent.getReferralType())
-            .provider(referralSent.getProviderCode())
-            .team(referralSent.getTeamCode())
-            .staff(referralSent.getStaffCode())
-            .date(referralSent.getDate())
-            .notes(referralSent.getNotes()).build();
-
-        return webClient.post()
-            .uri("/v1/contact")
-            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
-            .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
-            .bodyValue(contact)
-            .retrieve()
-            .toEntity(String.class)
-            .block();
+            .eventId(referralSent.getConvictionId())
+            .requirementId(referralSent.getRequirementId())
+            .referralDate(referralSent.getDate())
+            .status(referralSent.getNsiStatus())
+            .statusDate(referralSent.getDate().atStartOfDay())
+            .notes(referralSent.getNotes())
+            .intendedProvider(referralSent.getProviderCode())
+            .manager(NewNsiManager.builder()
+                .staff(referralSent.getStaffCode())
+                .team(referralSent.getTeamCode())
+                .provider(referralSent.getProviderCode())
+                .build()
+            ).build();
+
+        var newNsiReponse = deliusApiClient.createNewNsi(newNsiRequest);
+
+        return ReferralSentResponse.builder().nsiId(newNsiReponse.getId()).build();
+    }
+
+    public Optional<Nsi> getExistingMatchingNsi(String crn, ReferralSentRequest referralSent) {
+        // determine if there is an existing suitable NSI
+        var offenderId = offenderService.offenderIdOfCrn(crn).orElseThrow();
```

Comment:
maybe throw NotFoundException so resource returns 404 (although I suppose you have checked  for the offender before you get here)